homeassistant/components/accuweather/coordinator.py

```diff
@@ -0,0 +1,95 @@
+"""The AccuWeather coordinator."""
+
+from asyncio import timeout
+from datetime import timedelta
+import logging
+from typing import TYPE_CHECKING, Any
+
+from accuweather import AccuWeather, ApiError, InvalidApiKeyError, RequestsExceededError
+from aiohttp.client_exceptions import ClientConnectorError
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.update_coordinator import (
+    TimestampDataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import DOMAIN, MANUFACTURER
+
+EXCEPTIONS = (ApiError, ClientConnectorError, InvalidApiKeyError, RequestsExceededError)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AccuWeatherDataUpdateCoordinator(TimestampDataUpdateCoordinator[Any]):
```

Comment:
What's a TimestampDataUpdateCoordinator?

Reply:
This is a DataUpdateCoordinator which keeps track of the last successful update:
https://github.com/home-assistant/core/blob/c7e6f3696f16fc3a5626387cec5fac7e29d2b32f/homeassistant/helpers/update_coordinator.py#L453-L463

`WeaterEntity` uses it:
https://github.com/home-assistant/core/blob/c7e6f3696f16fc3a5626387cec5fac7e29d2b32f/homeassistant/components/weather/__init__.py#L147-L155